src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java

```diff
@@ -123,17 +122,24 @@ public void consume(Message<String> message) {
                 }
 
                 if (!processConfigs.isEmpty()) {
-                    // import IGM into the network store
-                    UUID networkUuid = caseFetcherService.importCase(caseUuid);
-                    // check IGM quality
-                    boolean valid = igmQualityCheckService.check(networkUuid);
+                    Mono<UUID> networkUuidMono = Mono.fromCallable(() -> caseFetcherService.importCase(caseUuid));
+
+                    Mono<Boolean> validMono = networkUuidMono.flatMap(networkUuid ->
+                       // check IGM quality
+                        Mono.fromCallable(() -> igmQualityCheckService.check(networkUuid))
+                    );
 
-                    merge(processConfigs.get(0), dateTime, date, tso, valid, networkUuid);
+                    validMono.zipWith(networkUuidMono, (valid, networkUuid) -> {
+                        merge(processConfigs.get(0), dateTime, date, tso, valid, networkUuid);
+                        return Mono.empty();
+                    }).subscribe();
```

Comment:
I don't think this is the right way to implement a spring reactive workflow. We are not supposed to subscribe here but only at the end of the method when the full task graph will be built.

Reply:
we shouldn't wait until the end of that merge process to begin the next merge processes that's why we launch the execution of that instruction by calling subscribe method